synapse/event_auth.py

```diff
@@ -64,28 +64,12 @@ def check(
     Returns:
          if the auth checks pass.
     """
-    assert isinstance(auth_events, dict)
-
     if do_size_check:
         _check_size_limits(event)
 
     if not hasattr(event, "room_id"):
         raise AuthError(500, "Event has no room_id: %s" % event)
 
-    room_id = event.room_id
-
-    # We need to ensure that the auth events are actually for the same room, to
-    # stop people from using powers they've been granted in other rooms for
-    # example.
-    for auth_event in auth_events.values():
-        if auth_event.room_id != room_id:
-            raise AuthError(
-                403,
-                "During auth for event %s in room %s, found event %s in the state "
-                "which is in room %s"
-                % (event.event_id, room_id, auth_event.event_id, auth_event.room_id),
-            )
-
```

Comment:
this (and the assertion above) are moved further down, into `check_auth_rules_for_event`. This could mean that we end up with a *different* `AuthError` for some bad events, but I don't think that matters: the important thing is that bad events are rejected and good events are accepted.